File: tests/test_tts.py
#!/usr/bin/env python3
"""Test TTS synthesis"""

import asyncio
from zena_mode.voice_manager import get_voice_manager


async def test_tts():
    """Test tts."""
    vm = get_voice_manager()

    print("Testing TTS Synthesis...\n")

    # Test 1: Simple synthesis
    print("1️⃣ Testing basic synthesis...")
    result = vm.synthesize("Hello, this is a test.")
    # [X-Ray auto-fix] print(f"   Success: {result['success']}")
    if result["success"]:
        # [X-Ray auto-fix] print(f"   Duration: {result['duration']} seconds")
        # [X-Ray auto-fix] print(f"   Audio data: {len(result['audio_data'])} chars (base64)")
        # [X-Ray auto-fix] print(f"   Has audio URL: {'audio_url' in result and bool(result['audio_url'])}")
        pass
    else:
        print(f"   Error: {result.get('error')}")

    # Test 2: Longer text
    print("\n2️⃣ Testing longer synthesis...")
    long_text = "Artificial Intelligence is transforming the world. From healthcare to education, AI is making systems smarter and more efficient."
    result = vm.synthesize(long_text)
    # [X-Ray auto-fix] print(f"   Success: {result['success']}")
    if result["success"]:
        # [X-Ray auto-fix] print(f"   Duration: {result['duration']} seconds")
        # [X-Ray auto-fix] print(f"   Audio data: {len(result['audio_data'])} chars")
        pass
    # Test 3: Cache test
    print("\n3️⃣ Testing cache...")
    result1 = vm.synthesize("Test cache", use_cache=True)
    result2 = vm.synthesize("Test cache", use_cache=True)
    # [X-Ray auto-fix] print(f"   First call success: {result1['success']}")
    # [X-Ray auto-fix] print(f"   Second call success: {result2['success']}")
    # [X-Ray auto-fix] print(f"   Same audio: {result1['audio_data'] == result2['audio_data']}")
    print("\n✅ All TTS tests passed!")


if __name__ == "__main__":
    asyncio.run(test_tts())

File: rust_output/src/test_tts.rs
/// Test TTS synthesis

use anyhow::{Result, Context};
use crate::voice_manager::{get_voice_manager};
use tokio;

/// Test tts.
pub async fn test_tts() -> () {
    // Test tts.
    let mut vm = get_voice_manager();
    println!("{}", "Testing TTS Synthesis...\n".to_string());
    println!("{}", "1️⃣ Testing basic synthesis...".to_string());
    let mut result = vm.synthesize("Hello, this is a test.".to_string());
    println!("   Success: {}", result["success".to_string()]);
    if result["success".to_string()] {
        println!("   Duration: {} seconds", result["duration".to_string()]);
        println!("   Audio data: {} chars (base64)", result["audio_data".to_string()].len());
        println!("   Has audio URL: {}", (result.contains(&"audio_url".to_string()) && (result["audio_url".to_string()] != 0)));
        // pass
    } else {
        println!("   Error: {}", result.get(&"error".to_string()).cloned());
    }
    println!("{}", "\n2️⃣ Testing longer synthesis...".to_string());
    let mut long_text = "Artificial Intelligence is transforming the world. From healthcare to education, AI is making systems smarter and more efficient.".to_string();
    let mut result = vm.synthesize(long_text);
    println!("   Success: {}", result["success".to_string()]);
    if result["success".to_string()] {
        println!("   Duration: {} seconds", result["duration".to_string()]);
        println!("   Audio data: {} chars", result["audio_data".to_string()].len());
        // pass
    }
    println!("{}", "\n3️⃣ Testing cache...".to_string());
    let mut result1 = vm.synthesize("Test cache".to_string(), /* use_cache= */ true);
    let mut result2 = vm.synthesize("Test cache".to_string(), /* use_cache= */ true);
    println!("   First call success: {}", result1["success".to_string()]);
    println!("   Second call success: {}", result2["success".to_string()]);
    println!("   Same audio: {}", result1["audio_data".to_string()] == result2["audio_data".to_string()]);
    println!("{}", "\n✅ All TTS tests passed!".to_string());
}
